x-pack/plugins/security_solution/public/common/components/toolbar/bulk_actions/helpers.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { EuiSelectableOption } from '@elastic/eui';
+import { intersection, union } from 'lodash';
+
+// Sorts in order of `on` -> `mixed` -> `undefined`
+const checkedSortCallback = (a: EuiSelectableOption, b: EuiSelectableOption) => {
+  if (a.checked) {
+    if (b.checked) {
+      return a.checked <= b.checked ? 1 : -1;
+    }
+    return -1;
+  }
+  if (b.checked) {
+    return 1;
+  }
+  return 0;
+};
+
+export const createInitialTagsState = (existingTags: string[][], defaultTags: string[]) => {
```

Comment:
this could be used with a reducer or even better in redux I think just based on the name alone. could also help prevent any potentially tricky issues where since the component this is used in renders more often than it should, this by extension runs more often than you intend. adding a console.log in the map here results in this map running 3 times when only 1 instance of BulkAlertTagsPanelComponent is rendered in the dom, probably don't need 2 of them. 

![create_initial_state_3_times](https://github.com/elastic/kibana/assets/56408403/0813a412-1580-4cee-9b2c-047f03ad4038)